auth_backends/backends.py

```diff
@@ -244,20 +244,33 @@ class EdXOAuth2(EdXBackendMixin, BaseOAuth2):
 
     @property
     def logout_url(self):
-        return '{}?client_id={}&redirect_url={}'.format(
-            self.end_session_url(),
-            self.setting('KEY'),
-            self.setting('LOGOUT_REDIRECT_URL')
-        )
+        if self.setting('LOGOUT_REDIRECT_URL'):
+            return '{}?client_id={}&redirect_url={}'.format(
+                self.end_session_url(),
+                self.setting('KEY'),
+                self.setting('LOGOUT_REDIRECT_URL')
+             )
+        else:
+            return self.end_session_url()
 
     def authorization_url(self):
-        return '{}/oauth2/authorize'.format(self.setting('URL_ROOT'))
+        url_root = self.setting('PUBLIC_URL_ROOT')
```

Comment:
Does `self.setting()` return None when setting not found for sure? Is that tested in tests below?